### PR TITLE
Fix undefined boolean options - Fix #988

### LINKED ISSUE #988: Boolean options always parsed as either true or false
Didn't have time to investigate too much, but it appears to me that option defined as boolean values but without explicit value are automatically parsed as false (rather then `undefined` which allowed to detect options who weren't passed explicitly)

### PR DESCRIPTION
**Pull request by SBoudrias**
Got to add and fix tests. But here's a first draft to tackle the bug.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -20,6 +20,8 @@ var Storage = require('./util/storage');
 var promptSuggestion = require('./util/prompt-suggestion');
 var debug = require('debug')('yeoman:generator');
 
+var EMPTY = '@@_YEOMAN_EMPTY_MARKER_@@';
```
Comment (mischah) [inline lib/index.js]: What’s `'@@_YEOMAN_EMPTY_MARKER_@@'`? 😳

Comment (SBoudrias) [inline lib/index.js]: I need a way to detect if an explicit value was passed or not. I though of using a `Symbol` given it'd be a guaranteed unique value, but minimist breaks if `default` is not a string or a number.

So, I went with a long string that hopefully won't collide with any real value

**Comment by SBoudrias** [comment]
Okay, I'll merge as is to unbreak 1.0 :)

**Comment by mischah** [comment]
😘

**Pull request merged**